eva/udfs/decorators/io_descriptors/data_types.py

```diff
@@ -41,6 +44,46 @@ def __init__(
             array_dimensions=dimensions,
         )
 
+    def check_array_and_convert_shape(self, input_object):
```

Comment:
Do we want to be converting data on our own? Or should that be an additional flag? For example user specifies that all data should be converted to this the given IOSignature format. We assert if it isn't possible

Reply:
I feel that it is redundant to add another flag given that we have check_input and check_output flags.